org.eclipse.xtend.ide.tests/src/org/eclipse/xtend/ide/tests/builder/JavaEditorExtension.xtend

```diff
@@ -100,13 +100,22 @@ class JavaEditorExtension {
 				}
 			], eventMask)
 		producer.apply
-		while (!changed.get) {
+		var retries = 30
```

Comment:
Can we refactor to remove the busy wait entirely?

```
Display display = Display.getCurrent();
assertNotNull(display);

AtomicBoolean stop = new AtomicBoolean();
display.timerExec(1000, ()->stop.set(true));

while (!changed.get && !stop.get && !display.readAndDispatch()) {
  display.sleep();
}
if (!changed.get) {
  throw new AssertionError(..)
}
```

Reply:
@szarnekow my proposal was also meant to remove the busy-wait since it sleeps a bit before retrying.

If I understand your proposal, that also removes the busy wait, but it only gives one second, right? Maybe that's too few?

However, yours is cleaner for sure :)

Reply:
Yeah, we can give it 10 seconds or similar. In most cases it will be faster, I guess.

Reply:
@szarnekow you meant something like that right?

```xtend
		var display = Display.getCurrent()
		assertNotNull(display)

		val stop = new AtomicBoolean()
		display.timerExec(10000, [stop.set(true)])

		while (!changed.get && !stop.get && display.readAndDispatch()) {
			if (VERBOSE) {
				println("Display sleep...")
			}
			display.sleep()
		}
		if (!changed.get) {
			throw new AssertionError("No event has been received")
		}
```

Note that it is `display.readAndDispatch()` in the while condition, not as you said `!display.readAndDispatch()` otherwise the loop exits immediately and the test fails.

However, with this solution, I see tons of `Display sleep...` so it looks like it's busy waiting again...

Reply:
Maybe:
```
		while (!changed.get && !stop.get) {
			if (!display.readAndDispatch()) {
				display.sleep();
			}
		}
```

Reply:
@szarnekow there are less "sleeping" messages, but still many in fast sequence.

The thread executing the code is not sleeping at all, I seem to understand

Reply:
The thread executing the code is the display thread. That thread should not sleep since it still must process the incoming events.
It's the regular SWT event loop code that we use here.

Reply:
Does it still turn on the fan of you machine when you run this test?

Reply:
@szarnekow I would say less than with the current code in `main` ;)

You're right about the Display thread should not sleep. My feeling was that by making it sleep the other threads could have more chances to do their job, but if these other threads are all related to the UI thread, then my proposal would make things worse.

I'll push force your proposal in my branch with verbose TRUE just to see in the CI how many messages it prints. Then I can remove verbosity and push force again so that we can merge if you're OK.